packages/@icon-magic/generate/src/plugins/svg-light-dark.ts

```diff
@@ -0,0 +1,172 @@
+
+    /*
+    - If `imageSet` exists and the `colorScheme` equals `dark, run the plugin.
+    - Plugin will skip the `light` flavors because their data can be collected through the `dark` flavor
+    */
+    if (imageSet && colorScheme === 'dark') {
+      // light flavor data
+      const lightFlavor = icon.flavors.get(imageSet);
+
+      // if light flavor doesn't exist, exit out of plugin
+      if (!lightFlavor) {
+        return flavor;
+      } else {
+        const darkFlavorContents = (await flavor.getContents()) as string; // .svg asset's getContents() returns a string (the svg as a string)
+        const lightFlavorContents  = (await lightFlavor.getContents()) as string;
+
+        const lightToken = params.lightToken;
+        const darkToken = params.darkToken;
+
+        const lightSvgo = new Svgo({
+          plugins: [
+            {
+              addAttributesToSVGElement: {
+                attributes: [{display: lightToken || "var(--svg-light-display)"}]
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        const darkSvgo = new Svgo({
+          plugins: [
+            {
+              addAttributesToSVGElement: {
+                attributes: [{display: darkToken || "var(--svg-dark-display)"}]
+              }
+            },
+            {
+              removeViewBox: false
+            }
+          ],
+          js2svg: { pretty: true, indent: 2 }
+        });
+
+        if (lightFlavorContents){
+          // create optimized light/dark svg assets
+          const lightAsset = await lightSvgo.optimize(lightFlavorContents); //returns a string
+          const darkAsset = await darkSvgo.optimize(darkFlavorContents);
+
+          const mixedParentSvg = await domSvgParentGenerate(lightAsset, darkAsset);
+
+          const mixedSvgo = new Svgo({
+            plugins: [
+              {
+                removeViewBox: false
+              }
+            ],
+            js2svg: { pretty: true, indent: 2 }
+          });
+
+          const mixedString = serializeToString(mixedParentSvg);
+          const mixedAsset = await mixedSvgo.optimize(mixedString); //returns a string
+
+          // write the parent svg with light/dark children svg's to the output directory
+          await fs.ensureDir(outputPath);
+          await fs.writeFile(
+            `${path.join(outputPath, imageSet)}-mixed.svg`,
+            mixedAsset.data,
+            {
+              encoding: 'utf8'
+            }
+          );
+
+          // Final new Mixed Flavor
+          const mixedFlavor: Flavor = new Flavor(icon.iconPath, {
+            name: `${imageSet}-mixed`,
+            path: `./${imageSet}-mixed.svg`,
+            colorScheme: 'mixed'
+          });
+
+          //Add new mixed flavor to icon.flavors. Is then processed during svg-generate and added to resulting iconrc.json file.
```

Comment:
```suggestion
          // Add new mixed flavor to icon.flavors. Is added to resulting iconrc.json file.
```

I don't think the plugins are guaranteed to run in order!

Reply:
i.e no guarantee svg-generate always runs right after this

Reply:
Ah ok, I was wondering about that, with no guarantee then we can add that. It does look to run this one first and then svgGenerate in my local runs, but I understand it's not a guarantee.